pkg/controller/funding_schedules.go

```diff
@@ -232,10 +239,31 @@ func (c *Controller) putFundingSchedules(ctx iris.Context) {
 
 			now := time.Now()
 			for _, spend := range spending {
-				if err := spend.CalculateNextContribution(c.getContext(ctx), account.Timezone, &request, now); err != nil {
+				funding, err := repo.GetSpendingFunding(c.getContext(ctx), bankAccountId, spend.SpendingId)
+				if err != nil {
+					c.wrapPgError(ctx, err, "failed to read funding instructions for spending")
+					return
+				}
+
+				updatedFunding, err := spend.CalculateNextContribution(
+					c.getContext(ctx),
+					account.Timezone,
+					models.NewSpendingFundingHelper(funding),
+					now,
+				)
+				if err != nil {
 					c.wrapAndReturnError(ctx, err, http.StatusInternalServerError, "failed to recalculate spending object")
 					return
 				}
+
+				if updatedFunding != nil {
+					if err = repo.UpdateSpendingFunding(c.getContext(ctx), bankAccountId, []models.SpendingFunding{
+						*updatedFunding,
+					}); err != nil {
+						c.wrapPgError(ctx, err, "failed to update spending funding")
+						return
+					}
```

Comment:
There are a ton of these, maybe I should just make a helper function that takes a single spending funding object that is nullable?

And if nil is passed to it, then the function is a no-op; but if something is passed to it then it does the update?